### PR TITLE
Fos user bundle integration

### PR DESCRIPTION
**Pull request by pueppiblue**
Also includes some style improvements.
Adds the need for some more test for additional roles. As of now only ANONYMOUS some routes are tested.
You may also want to check the Authentication_Checker service injection. Great to check for user roles inside the MenuBuilder.

### CODE REVIEW

File: app/DoctrineMigrations/Version20160803183748.php [hunk 1]
```diff
@@ -8,19 +8,20 @@
 /**
  * Auto-generated Migration: Please modify to your needs!
  */
-class Version20160729173849 extends AbstractMigration
+class Version20160803183748 extends AbstractMigration
```
Comment (robertfausk) [inline app/DoctrineMigrations/Version20160803183748.php]: Do not overwrite migrations. Better create new ones to get a feeling of how to handle a lot of migrations.

File: app/DoctrineMigrations/Version20160803183748.php [hunk 2]
```diff
@@ -31,11 +32,12 @@ public function up(Schema $schema)
     public function down(Schema $schema)
     {
         // this down() migration is auto-generated, please modify it to your needs
-        $this->abortIf($this->connection->getDatabasePlatform()->getName() !== 'mysql', 'Migration can only be executed safely on \'mysql\'.');
+        $this->abortIf($this->connection->getDatabasePlatform()->getName() != 'mysql', 'Migration can only be executed safely on \'mysql\'.');
```
Comment (robertfausk) [inline app/DoctrineMigrations/Version20160803183748.php]: Why did you change comparison from identical to equal?
Always prefer identical to equal.

File: app/Resources/FOSUserBundle/views/Registration/confirmed.html.twig [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+
+{% block fos_user_content %}
+    <div class="card">
+        <div class="card-content">
+            <span class="card-title col s12">
+                {{ 'layout.register'|trans }}
+            </span>
+            <p>{{ 'registration.confirmed'|trans({'%username%': user.username}) }}</p>
+            {% if targetUrl %}
+            <p><a href="{{ targetUrl }}">{{ 'registration.back'|trans }}</a></p>
```
Comment (robertfausk) [inline app/Resources/FOSUserBundle/views/Registration/confirmed.html.twig]: indent

File: app/Resources/FOSUserBundle/views/Registration/register_content.html.twig [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+
+<div class="card">
+    <div class="card-content">
+        <div class="card-title col s12">
+            <h4>{{ 'layout.register'|trans }}</h4>
+        </div>
+        <div class="row">
+            {{ form_start(form, {'method': 'post', 'action': path('fos_user_registration_register'), 'attr': {'class': 'col s12 fos_user_registration_register'}}) }}
+            <div class="row">
+                {{ form_row(form.email, { 'attr' : {'class' : 'validate'}}) }}
```
Comment (robertfausk) [inline app/Resources/FOSUserBundle/views/Registration/register_content.html.twig]: whitespace

File: app/Resources/FOSUserBundle/views/Registration/register_content.html.twig [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+            </div>
+            <div class="row">
+                {{ form_row(form.username, { 'attr' : {'class' : 'validate'}}) }}
```
Comment (robertfausk) [inline app/Resources/FOSUserBundle/views/Registration/register_content.html.twig]: whitespace

File: app/Resources/FOSUserBundle/views/Registration/register_content.html.twig [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+            </div>
+            <div class="row">
+                {{ form_row(form.plainPassword, { 'attr' : {'class': 'validate'}}) }}
```
Comment (robertfausk) [inline app/Resources/FOSUserBundle/views/Registration/register_content.html.twig]: whitespace

File: app/Resources/FOSUserBundle/views/layout.html.twig [hunk 7]
```diff
@@ -0,0 +1,16 @@
+{% extends 'base.html.twig' %}
+
+{% block title %}
+    {{ 'add translation here' }}
+{% endblock %}
+
+{% block body %}
+    <br>
```
Comment (robertfausk) [inline app/Resources/FOSUserBundle/views/layout.html.twig]: Don't use ```<br>``` for styling purposes.

File: app/Resources/views/User/listUnordered.html.twig [hunk 8]
```diff
@@ -11,10 +11,10 @@
         <div class="col s12 card-panel">
             <ul class="row">
                 {% for user in users %}
-                    <li class="col s12 m3 l2">
+                    <li class="col s4 m2 l1">
                         <a href="{{ path('user_show', {'id': user.id}) }}">
                             <div class="multi-panel card-panel">
-                                {{ user.getAccountName }}
+                                    {{ user.getAccountName }}
```
Comment (robertfausk) [inline app/Resources/views/User/listUnordered.html.twig]: indent

File: app/Resources/views/User/listCharacters.html.twig [hunk 9]
```diff
@@ -1,43 +1,27 @@
 <div class="card">
     <div class="card-content">
-        <span class="card-title">
+        <div class="col s12 card-title">
             <h5>Verknüpfte Charaktere:</h5>
```
Comment (robertfausk) [inline app/Resources/views/User/listCharacters.html.twig]: Check heading level. h5 is very deep. I would assume h2.

File: app/Resources/views/User/listCredits.html.twig [hunk 10]
```diff
@@ -1,51 +1,29 @@
 <div class="card">
     <div class="card-content">
-        <span class="card-title">
+        <div class="card-title col s12">
             <h5>Derzeitige Listenpunkte:</h5>
```
Comment (robertfausk) [inline app/Resources/views/User/listCredits.html.twig]: Check heading level. h5 is very deep. I would assume h2.

File: app/Resources/views/navbar.html.twig [hunk 11]
```diff
@@ -1,10 +1,14 @@
+{% trans_default_domain 'FOSUserBundle' %}
+
 <nav>
     <div class="nav-wrapper">
-        <a href="#!" class="brand-logo right"><img style="max-height: 2em" src="{{ asset('images/aoc_logo1.jpg') }}"></a>
+        {#<a href="#!" class="brand-logo right right-align"><img style="max-height: 2em" src="{{ asset('images/aoc_logo1.jpg') }}"></a>#}
```
Comment (robertfausk) [inline app/Resources/views/navbar.html.twig]: remove unused code

File: src/AppBundle/Menu/MenuBuilder.php [hunk 12]
```diff
@@ -44,18 +46,47 @@ public function createNavMenu()
 
     }
 
-    public function createSideMenu()
+    public function createUserMenu(array $options)
+    {
+        $isAdmin = $this->checker->isGranted('ROLE_ADMIN');
+        $isUser = $this->checker->isGranted('ROLE_USER');
+        $loggedIn = ($isAdmin || $isUser);
```
Comment (robertfausk) [inline src/AppBundle/Menu/MenuBuilder.php]: check variable name

File: src/AppBundle/Menu/MenuBuilder.php [hunk 13]
```diff
@@ -44,18 +46,47 @@ public function createNavMenu()
...
+
+        $menu = $this->factory->createItem('root');
+        $menu->setChildrenAttribute('class', 'right hide-on-med-and-down')
+            ->setExtra('translation_domain', 'FOSUserBundle');
+
+        if ($loggedIn) {
+            /** @var User $userName */
```
Comment (robertfausk) [inline src/AppBundle/Menu/MenuBuilder.php]: check variable name type hint

File: src/AppBundle/Menu/MenuBuilder.php [hunk 14]
```diff
@@ -44,18 +46,47 @@ public function createNavMenu()
 
     }
 
-    public function createSideMenu()
+    public function createUserMenu(array $options)
```
Comment (robertfausk) [inline src/AppBundle/Menu/MenuBuilder.php]: add doc block

File: src/AppBundle/Menu/MenuBuilder.php [hunk 15]
```diff
@@ -15,14 +16,15 @@ class MenuBuilder
      * @var FactoryInterface
      */
     private $factory;
-
+    private $checker;
 
     /**
```
Comment (robertfausk) [inline src/AppBundle/Menu/MenuBuilder.php]: tpye hint missing in method doc block

File: src/AppBundle/Menu/MenuBuilder.php [hunk 16]
```diff
@@ -44,18 +46,47 @@ public function createNavMenu()
...
+            $menu->addChild('Login', ['route' => 'fos_user_security_login'])
+                ->setExtra('translation_domain', 'FOSUserBundle');
+
+            $menu->addChild('Register', ['route' => 'fos_user_registration_register']);
+        }
+
+        return $menu;
+    }
+
+    public function createSideMenu($options)
```
Comment (robertfausk) [inline src/AppBundle/Menu/MenuBuilder.php]: add doc block

File: app/DoctrineMigrations/Version20160803183748.php (hunk 1 above)
Comment (pueppiblue) [inline app/DoctrineMigrations/Version20160803183748.php]: I can do that in the future. I think its only necessary once an app is in production, but to learn we can do this for development too.

File: app/DoctrineMigrations/Version20160803183748.php (hunk 2 above)
Comment (pueppiblue) [inline app/DoctrineMigrations/Version20160803183748.php]: Sure. I did not touch any migration sql. Only generating them via console's doctrine:migrate.

File: app/Resources/FOSUserBundle/views/Registration/register_content.html.twig (hunk 4 above)
Comment (pueppiblue) [inline app/Resources/FOSUserBundle/views/Registration/register_content.html.twig]: I would appreciate a more precise feedback.
Indentation whitespace, whitespace in parameter declaration or missing whitespace somewhere?

File: app/Resources/FOSUserBundle/views/layout.html.twig (hunk 7 above)
Comment (pueppiblue) [inline app/Resources/FOSUserBundle/views/layout.html.twig]: I added padding-top to container class instead.
If you have a better/alternate solution, i'm all ears.

File: app/Resources/views/User/listCharacters.html.twig (hunk 9 above)
Comment (pueppiblue) [inline app/Resources/views/User/listCharacters.html.twig]: <h2> is very big.
Could use it though and change text size by css.

File: app/DoctrineMigrations/Version20160803183748.php (hunk 1 above)
Comment (robertfausk) [inline app/DoctrineMigrations/Version20160803183748.php]: I prefer your sugggestion :+1:

File: app/DoctrineMigrations/Version20160803183748.php (hunk 2 above)
Comment (robertfausk) [inline app/DoctrineMigrations/Version20160803183748.php]: Ah ok. Wasn't sure about this. Maybe it's because of doctrine update. Doesn't really matter.

File: app/Resources/FOSUserBundle/views/Registration/register_content.html.twig (hunk 4 above)
Comment (robertfausk) [inline app/Resources/FOSUserBundle/views/Registration/register_content.html.twig]: Sorry for that. I didn't recognize that my comment wasn't on 11th line. I will be more concrete next time.

File: app/Resources/FOSUserBundle/views/layout.html.twig (hunk 7 above)
Comment (robertfausk) [inline app/Resources/FOSUserBundle/views/layout.html.twig]: Maybe give some distance to header. I'm not really sure what you want to achieve and where you want it. We can talk about it. It's ok for now.

File: app/Resources/views/User/listCharacters.html.twig (hunk 9 above)
Comment (robertfausk) [inline app/Resources/views/User/listCharacters.html.twig]: You can use something like ```<h2 class="h5">...```
Also see http://v4-alpha.getbootstrap.com/content/typography/#headings

**Comment by robertfausk** [comment]
You can merge and close it if you want. Looks good :+1:
